pkg/handlers/app.go

```diff
@@ -251,38 +268,44 @@ func (h *Handler) ListApps(w http.ResponseWriter, r *http.Request) {
 		}
 
 		for _, app := range cache {
-			responseApps = append(responseApps, *app.Application)
+			helmResponseApps = append(helmResponseApps, *app.Application)
 		}
-	} else {
-		apps, err := store.GetStore().ListInstalledApps()
-		if err != nil {
-			logger.Error(err)
-			w.WriteHeader(http.StatusInternalServerError)
-			return
+
+		listAppsResponse := types.ListAppsHelmResponse{
+			Apps: helmResponseApps,
 		}
 
-		defaultRoles := rbac.DefaultRoles() // TODO (ethan): this should be set in the handler
+		JSON(w, http.StatusOK, listAppsResponse)
+		return
+	}
+	apps, err := store.GetStore().ListInstalledApps()
+	if err != nil {
+		logger.Error(err)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
 
-		for _, a := range apps {
-			if sess.HasRBAC { // handle pre-rbac sessions
-				allow, err := rbac.CheckAccess(r.Context(), defaultRoles, "read", fmt.Sprintf("app.%s", a.Slug), sess.Roles)
-				if err != nil {
-					logger.Error(errors.Wrapf(err, "failed to check access for app %s", a.Slug))
-					w.WriteHeader(http.StatusInternalServerError)
-					return
-				} else if !allow {
-					continue
-				}
-			}
+	defaultRoles := rbac.DefaultRoles() // TODO (ethan): this should be set in the handler
```

Comment:
The diff from here down is solely caused by moving this code out of an `else` block since the above `if` now returns on its own.